kie-pmml-trusty/kie-pmml-models/kie-pmml-models-scorecard/kie-pmml-models-scorecard-tests/src/test/java/org/kie/pmml/models/scorecard/tests/SimpleScorecardCategoricalTest.java

```diff
@@ -81,4 +86,25 @@ public void testSimpleScorecardCategorical() {
         Assertions.assertThat(pmml4Result.getResultVariables().get(REASON_CODE1_FIELD)).isEqualTo(reasonCode1);
         Assertions.assertThat(pmml4Result.getResultVariables().get(REASON_CODE2_FIELD)).isEqualTo(reasonCode2);
     }
+
+    @Test
+    public void testSimpleScorecardCategoricalVerifyNoException() {
+        final List<Map<String, Object>> samples = getSamples();
+        samples.forEach(inputData -> {
+            PMML4Result retrieved = evaluate(pmmlRuntime, inputData, MODEL_NAME);
+            assertNotNull(retrieved);
+        });
+    }
+
+    private List<Map<String, Object>> getSamples() {
+        final List<Map<String, Object>> toReturn = new ArrayList<>();
+        for (int i = 0; i < 10; i++) {
+            final Map<String, Object> toAdd = new HashMap<>();
+            toAdd.put("input1", CATEGORY[i % CATEGORY.length]);
+            toAdd.put("input2", CATEGORY[Math.abs(CATEGORY.length - i) % CATEGORY.length]);
+            toReturn.add(toAdd);
```

Comment:
```suggestion
            toReturn.add(new HashMap<String, Object>() {{
                put("input1", CATEGORY[i % CATEGORY.length]);
                put("input2", CATEGORY[Math.abs(CATEGORY.length - i) % CATEGORY.length]);
            }});
```
the rest of the for block disappears

Reply:
the same for SimpleScorecardCategoricalTest.java

Reply:
@Ginxo 
Can't do that because inside the Lambda variable `i` is not final...

Reply:
@gitgabrio you are right. I don't want to complicate things but we can additionally do something like
```
private List<Map<String, Object>> getSamples() {
        return IntStream.range(0, 10).boxed().map(i -> new HashMap<String, Object>() {{
            put("input1", CATEGORY[i % CATEGORY.length]);
            put("input2", CATEGORY[Math.abs(CATEGORY.length - i) % CATEGORY.length]);
        }}).collect(Collectors.toList());
}
```            
`IntStream.range(0, 15)

Reply:
just to clarify, I'm totally ok with this PR, I'm just trying to provide different ways to simplify code, I hope you understand the reason why for my proposal @gitgabrio :heart:

Reply:
@Ginxo 
Done, thanks! (as for the former suggestion, I've done in the 7.x  PR)

Reply:
@Ginxo 
I appreciate your suggestions, I like them :smile: 
TBH, I simply copy-and-pasted code from some test failing inside kogito-apps, as-it-is, not looking for any kind of improvement; but your suggestions make sense to me, thanks!